### DISCUSSION TITLE
Why require Node 20.15.1 and up in v38 ?

### DISCUSSION THREAD

**Discussion by ddolcimascolo** [renovatebot/renovate#30376] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

When upgrading to v38, Renovate fails with 

```
ERROR: Unsupported node environment detected. Please update your node version.
       "versions": {
         "node": "20.9.0",
         "acorn": "8.10.0",
         "ada": "2.6.0",
         "ares": "1.19.1",
         "base64": "0.5.0",
         "brotli": "1.0.9",
         "cjs_module_lexer": "1.2.2",
         "cldr": "[43](https://gitlab.stordata.fr/stordata/renovate-bot/-/jobs/517672#L43).1",
         "icu": "73.2",
         "llhttp": "8.1.1",
         "modules": "115",
         "napi": "9",
         "nghttp2": "1.57.0",
         "nghttp3": "0.7.0",
         "ngtcp2": "0.8.1",
         "openssl": "3.0.10+quic",
         "simdutf": "3.2.17",
         "tz": "2023c",
         "undici": "5.26.3",
         "unicode": "15.0",
         "uv": "1.46.0",
         "uvwasi": "0.0.18",
         "v8": "11.3.2[44](https://gitlab.stordata.fr/stordata/renovate-bot/-/jobs/517672#L44).8-node.16",
         "zlib": "1.2.13.1-motley"
       },
       "range": "^20.15.1"
```

I'm on Node 20.9.0, Renovate should run just fine...

Can you elaborate on this very specific Node version requirement ?

Thanks,
David

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Chosen because of https://nodejs.org/en/blog/vulnerability/july-2024-security-releases

**Reply by ddolcimascolo** [reply]
While I do understand the rationale, you're trying to be too smart here... Renovate should only mandate the minimum version required for its execution and not anything else. That's what the `engines` fields in package.json is here for

**Reply by viceice** [reply]
yes, and we decided to limit supported versions.

**Comment by ErikTempelaarVO** [comment]
I found this issue because my azure pipeline running renovate broke with a similar error.
First I tried running an older version of renovate, but I upgrade my runner to 24.04 instead.

```
  - bash: |
      npx renovate@37.440.7
```

```
pool:
  vmImage: ubuntu-24.04
```

https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md

I didn't find this new version hosted-agent at first, so maybe this helps anyone. (-latest still points to 22.04)

**Reply by HonkingGoose** [reply]
The `github/actions/runner-images` readme says the Ubuntu 24.04 (Noble) image is in _beta_. That's why `ubuntu-latest` points to 22.04 (Jammy).

**Reply by seggcsont** [reply]
For me ubuntu-24.04 is not available. My solution was to upgrade Nodejs:
```
- task: NodeTool@0
  inputs:
    versionSource: 'spec'
    versionSpec: '20.x'
```

**Reply by HonkingGoose** [reply]
This info only applies to those who self-host Renovate, and don't use our Docker images.

Here's what comes with each GitHub Actions runner image:

- The [`ubuntu-24.04` image comes with Node.js `20.15.1`](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md)
- The [`ubuntu-22.04` image comes with Node.js `18.20.4`](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md), this is the current `ubuntu-latest` image

This means that if you use the _default_ configuration, you must install Node.js 20 into the image. 

If you don't want to install things, you can force the `ubuntu-24.04` image in your GitHub Actions workflow files.
